security/symmetric/src/symmetric_key.ts

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+import { Callback, callbackToPromise, SharedAccessSignature, anHourFromNow } from 'azure-iot-common';
+import { ArgumentError } from 'azure-iot-common/lib/errors';
+
+/**
+ * @private
+ */
+export class SymmetricKeySecurityClient  {
+
+  private _registrationId: string;
+  private _symmetricKey: string;
+  private _sas: SharedAccessSignature;
+
+  constructor(symmetricKey: string, registrationId: string) {
+    this._symmetricKey = symmetricKey;
+    this._registrationId = registrationId;
+  }
+
+
+  /**
+   * @method           module:azure-iot-security-symmetric-key.SymmetricKeySecurityClient#getRegistrationId
+   * @description      Returns the registrationId originally provided to the client.
+   * @param {function}          callback        Invoked upon completion of the operation.
+   *                                            If the err argument is non-null then the registrationId
+   *                                            parameter will be undefined.
+   */
+  getRegistrationId(callback?: Callback<string>): Promise<string> | void {
```

Comment:
>getRegistrationId [](start = 2, length = 17)

I see no reason for this to be asynchronous. same for createAuthenticationToken. unless we're building an API where we expect customers to recreate symmetric key clients?

Reply:
I disagree.  There is no additional work on the clients.  The installs simply happen.
Api consistency is something that developers value.

---
In reply to: [231755879](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r231755879) [](ancestors = 231755879,231752811)

Reply:
doesnt' address the need for this to be async (there isn't as far as I can tell). 

---
In reply to: [232004651](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r232004651) [](ancestors = 232004651,231755879,231752811)

Reply:
irl agreed fine.

---
In reply to: [232119149](https://github.com/Azure/azure-iot-sdk-node/pull/418#discussion_r232119149) [](ancestors = 232119149,232004651,231755879,231752811)